opensearch-project/OpenSearch-Dashboards
# Fix add to dashboard after saving

### Description

When creating a new visualization from a dashboard, and saving it with the "Add to Dashboards after saving" toggle checked, adding the newly saved visualization to the dashboard can fail. This is easily reproducible when `storeInSessionStorage` is enabled.

The way it is supposed to work is that the scopedHistory receives the ID of the newly create visualization in the current location state, and the dashboard container uses that to add the visualization. However, when the dashboard application state is created, it can update the URL at startup, which causes the location state to be lost. This fix preserves the current scopedHistory location state across that URL update.

### Issues Resolved

- fix: Fix add to dashboard after saving 

## Screenshot

N/A

## Testing the changes

1. Enable `storeInSessionStorage`
2. Go to an existing dashboard and click Edit
3. Click Create New
4. Pick a visualization, make whatever changes are desired, then click Save
5. Ensure "Add to Dashboards after saving" is enabled in the modal, then click Save and Return
6. The new visualization should be added to the dashboard. This would fail before this change.

## Changelog
- fix: Preserve location state at dashboard app startup to fix adding a new visualization

### Check List
